File: user_info/src/main/java/com/eapashkov/user_info/service/implement/UserInfoServiceImpl.java
package com.eapashkov.user_info.service.implement;

import com.eapashkov.user_info.feign.DownloadServiceFeignClient;
import com.eapashkov.user_info.model.UserInfoFromWorkSheet;
import com.eapashkov.user_info.repository.UserInfoFromWorkSheetRepository;
import com.eapashkov.user_info.service.UserInfoService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.core.io.ByteArrayResource;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.JAXBException;
import java.io.IOException;

@Service
@Slf4j
@AllArgsConstructor
@Transactional
public class UserInfoServiceImpl implements UserInfoService {

    private final UserInfoFromWorkSheetRepository userInfoFromWorkSheetRepository;
    private final DownloadServiceFeignClient downloadServiceFeignClient;


    public void saveUserInfoFromXml(String id) throws JAXBException, IOException {

        ByteArrayResource xmlFromDownloadService = downloadServiceFeignClient.getXMLUserInfo(id);
        UserInfoFromWorkSheet userInfoFromWorkSheet = unmarshalToModel(xmlFromDownloadService);
        userInfoFromWorkSheetRepository.save(userInfoFromWorkSheet);

        log.info("User with ID: {} has been saved to DB", userInfoFromWorkSheet.getId());

    }

    private UserInfoFromWorkSheet unmarshalToModel(ByteArrayResource resource) throws IOException, JAXBException {
        JAXBContext jaxbContext = JAXBContext.newInstance(UserInfoFromWorkSheet.class);

        return (UserInfoFromWorkSheet) jaxbContext.createUnmarshaller()
                .unmarshal(resource.getInputStream());

    }


}

File: user_info/src/main/java/com/eapashkov/user_info/service/implement/UserInfoServiceImpl.kt
package com.eapashkov.user_info.service.implement

import com.eapashkov.user_info.feign.DownloadServiceFeignClient
import com.eapashkov.user_info.model.UserInfoFromWorkSheet
import com.eapashkov.user_info.repository.UserInfoFromWorkSheetRepository
import com.eapashkov.user_info.service.UserInfoService
import mu.KotlinLogging
import org.springframework.core.io.ByteArrayResource
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.io.IOException
import javax.xml.bind.JAXBContext
import javax.xml.bind.JAXBException

@Service
@Transactional
class UserInfoServiceImpl(
    private val userInfoFromWorkSheetRepository: UserInfoFromWorkSheetRepository,
    private val downloadServiceFeignClient: DownloadServiceFeignClient
) : UserInfoService {

    private val logger = KotlinLogging.logger {}


    @Throws(JAXBException::class, IOException::class)
    override fun saveUserInfoFromXml(id: String?) {

        val xmlFromDownloadService = downloadServiceFeignClient.getXMLUserInfo(id)
        val userInfoFromWorkSheet = xmlFromDownloadService?.let { unmarshalToModel(it) }
        if (userInfoFromWorkSheet != null) {
            userInfoFromWorkSheetRepository.save(userInfoFromWorkSheet)
        }

        logger.info("User with ID: {} has been saved to DB", userInfoFromWorkSheet)
    }

    @Throws(IOException::class, JAXBException::class)
    private fun unmarshalToModel(resource: ByteArrayResource): UserInfoFromWorkSheet {
        val jaxbContext = JAXBContext.newInstance(UserInfoFromWorkSheet::class.java)
        return jaxbContext.createUnmarshaller()
            .unmarshal(resource.inputStream) as UserInfoFromWorkSheet
    }
}
